hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +78,945 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
+
+        // get accumulated momentum for particle
+        const Scalar3 linmom_accum(h_linmom_accum.data[central_idx]);
+        const vec3<Scalar> angmom_accum(h_angmom_accum.data[central_idx]);
+
+        // compute and store new velocity
+        Scalar4 vel_mass = h_velocity.data[central_idx];
+        const Scalar mass = vel_mass.w;
+        if (mass > 0)
+            {
+            vel_mass.x += linmom_accum.x / mass;
+            vel_mass.y += linmom_accum.y / mass;
+            vel_mass.z += linmom_accum.z / mass;
+            h_velocity.data[central_idx] = vel_mass;
+            }
+
+        // compute new angular momentum
+        quat<Scalar> angmom(h_angmom.data[central_idx]);
+        const quat<Scalar> orientation(h_orientation.data[central_idx]);
+
+        // convert angular momentum to quaternion and update
+        const vec3<Scalar> inertia(h_inertia.data[central_idx]);
+        vec3<Scalar> angmom_change_body = rotate(conj(orientation), angmom_accum);
+        if (inertia.x == Scalar(0))
+            {
+            angmom_change_body.x = Scalar(0);
+            }
+
+        if (inertia.y == Scalar(0))
+            {
+            angmom_change_body.y = Scalar(0);
+            }
+
+        if (inertia.z == Scalar(0))
+            {
+            angmom_change_body.z = Scalar(0);
+            }
+        angmom += Scalar(2.0) * orientation * quat(0.0, angmom_change_body);
+
+        // save update
+        h_angmom.data[central_idx] = quat_to_scalar4(angmom);
+        }
+    }
+
+#ifdef ENABLE_HIP
+//! Create autotuners
+void mpcd::CollisionMethod::checkRigidAutotuners()
+    {
+    const bool rigid_body_collision = m_embed_group && m_rigid_bodies;
+    std::vector<std::shared_ptr<AutotunerBase>> new_autotuners;
+    new_autotuners.insert(new_autotuners.end(),
+                          {m_drawrandvec_tuner,
+                           m_netvelo_tuner,
+                           m_applyrandvec_tuner,
+                           m_store_tuner,
+                           m_accumulate_tuner,
+                           m_transfer_tuner});
+    if (m_exec_conf->isCUDAEnabled() && rigid_body_collision)
+        {
+        // add tuners if they aren't already there
+        for (std::shared_ptr<AutotunerBase> rigid_tuner : new_autotuners)
+            {
+            if (std::find(m_autotuners.begin(), m_autotuners.end(), rigid_tuner)
+                == m_autotuners.end())
+                {
+                m_autotuners.push_back(rigid_tuner);
+                }
+            }
+        }
+    else
+        {
+        // remove rigid tuners if present
+        for (unsigned int t = 0; t < m_autotuners.size(); ++t)
+            {
+            if (std::find(new_autotuners.begin(), new_autotuners.end(), m_autotuners[t])
+                != new_autotuners.end())
+                {
+                m_autotuners.erase(m_autotuners.begin() + t);
+                }
+            }
+        }
```

Comment:
This seems problematic to me because the size of the autotuners vector will change if you erase from it. Can't you loop over the objects in `new_autotuners`, then do a find and erase? Something like:
```suggestion
        // remove rigid tuners if present
        for (auto& rigid_tuner : new_autotuners)
            {
            auto it = std::find(m_autotuners.begin(), m_autotuners.end(), rigid_tuner);
            if (it != m_autotuners.end())
                {
                m_autotuners.erase(it);
                }
            }
        }
```